Mathlib/Algebra/Homology/HomotopyCategory/KInjective.lean

```diff
@@ -0,0 +1,111 @@
+/-
+Copyright (c) 2025 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+module
+
+public import Mathlib.Algebra.Homology.Embedding.CochainComplex
+public import Mathlib.Algebra.Homology.HomotopyCategory.HomComplexInduction
+public import Mathlib.Algebra.Homology.Homotopy
+
+/-!
+# K-injective cochain complexes
+
+We define the notion of K-injective cochain complex in an abelian category,
+and show that bounded below complexes of injective objects are K-injective.
+
+## TODO (@joelriou)
+* Show that a cochain complex is K-injective iff its image in the homotopy
+category belongs to the right orthogonal of the triangulated subcategory
+of acyclic complexes
+* Deduce that we can compute morphisms to a K-injective complex in the
+derived category as homotopy classes of morphisms
+* Provide an API for computing `Ext`-groups using an injective resolution
+* Dualize everything
+
+## References
+* [N. Spaltenstein, *Resolutions of unbounded complexes*][spaltenstein1998]
+
+-/
+
+@[expose] public section
+
+namespace CochainComplex
+
+open CategoryTheory Limits HomComplex Preadditive
+
+variable {C : Type*} [Category C] [Abelian C]
+
+/-- A cochain complex `L` is K-injective if any morphism `K ⟶ L`
+with `K` acyclic is homotopic to zero. -/
+class IsKInjective (L : CochainComplex C ℤ) : Prop where
+  nonempty_homotopy_zero {K : CochainComplex C ℤ} (f : K ⟶ L) :
+    K.Acyclic → Nonempty (Homotopy f 0)
```

Comment:
Spaltenstein defines this as the hom complex from `K` to `L` beging acyclic for every acyclic `K`. Are these equivalent and if so, can we formalize the equivalence? What are the arguments for choosing this definition?